extensions/functions_arithmetic.yaml

```diff
@@ -441,6 +441,57 @@ aggregate_functions:
         decomposable: MANY
         intermediate: fp64?
         return: fp64?
+  - name: "product"
+    description: Product of a set of values.
+    impls:
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
+            required: false
+          - value: i8
+        nullability: MIRROR
```

Comment:
To follow the result of the sum discussion, this should be `DECLARED_OUTPUT`.

Reply:
Should be okay now.